### PR TITLE
12427 Removed staff payment component from conversion filing.

### PR DESCRIPTION
**Pull request by lambert-alex**
*Issue #:* /bcgov/[entity12427](https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/bcgov/entity/12427)

*Description of changes:*

- Removed staff payment component and associated methods from Conversion view.
- Updated ResourceIF to use FilingDataIF
- Hardcoded waive fees code to for a GP and SP conversion filing.
- Updated Alteration (BenefitCompanyResource, CooperativeResource), 
   Change (GeneralPartnershipResource, SoleProprietorshipResource) and
   Conversion (GeneralPartnershipRescource, SoleProprietorshipResource)
   to use the updated ResourceIF.
- App version = 3.1.13

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/resources/Change/GeneralPartnershipResource.ts [hunk 1]
```diff
@@ -8,10 +8,11 @@ export const GeneralPartnershipResource: ResourceIF = {
   displayName: GetCorpFullDescription(CorpTypeCd.PARTNERSHIP),
   nameRequestType: NameRequestEntityTypes.GP,
   addressLabel: 'Business Addresses',
-  filingData: {
+  filingData: [{
+    filingTypeCode: FilingCodes.CONVERSION,
     entityType: CorpTypeCd.PARTNERSHIP,
-    filingTypeCode: FilingCodes.CHANGE_OF_REGISTRATION
-  },
+    priority: false
```
Comment (lambert-alex) [inline src/resources/Change/GeneralPartnershipResource.ts:14]: priority is a required prop in FilingDataIF and would get compile warning if I did not add them. So defaulted them to false.

File: src/resources/Change/SoleProprietorshipResource.ts [hunk 2]
```diff
@@ -8,10 +8,11 @@ export const SoleProprietorshipResource: ResourceIF = {
   displayName: GetCorpFullDescription(CorpTypeCd.SOLE_PROP),
   nameRequestType: NameRequestEntityTypes.FR,
   addressLabel: 'Business Addresses',
-  filingData: {
-    entityType: CorpTypeCd.SOLE_PROP,
-    filingTypeCode: FilingCodes.CHANGE_OF_REGISTRATION
-  },
+  filingData: [{
+    filingTypeCode: FilingCodes.CONVERSION,
```
Comment (severinbeauvais) [inline src/resources/Change/SoleProprietorshipResource.ts:12]: Wrong filing code. Should be CHANGE_OF_REGISTRATION.

File: src/resources/Alteration/CooperativeResource.ts [hunk 3]
```diff
@@ -8,11 +8,11 @@ export const CooperativeResource: ResourceIF = {
   displayName: GetCorpFullDescription(CorpTypeCd.COOP),
   nameRequestType: NameRequestEntityTypes.CP,
   addressLabel: 'Registered Office',
-  filingData: {
-    entityType: CorpTypeCd.COOP,
-    filingTypeCode: FilingCodes.ALTERATION,
+  filingData: [{
+    filingTypeCode: FilingCodes.CONVERSION,
```
Comment (severinbeauvais) [inline src/resources/Alteration/CooperativeResource.ts:12]: Wrong filing code. Should be ALTERATION.
Wrong entity type. Should be COOP.

Actually, just check all the ones you changed. Each resource file is specific to a filing type and entity type (which I can see by the path).

File: src/interfaces/resource-interfaces/resource-interface.ts [hunk 4]
```diff
@@ -9,11 +8,7 @@ export interface ResourceIF {
   displayName: string
   nameRequestType: NameRequestEntityTypes
   addressLabel: string
-  filingData: {
-    entityType: CorpTypeCd
-    filingTypeCode: FilingCodes
-    priority?: boolean
-  }
+  filingData: Array<FilingDataIF>
```
[the remark this answers is not in the document]
Comment (lambert-alex) [reply src/interfaces/resource-interfaces/resource-interface.ts]: I fixed that.

File: src/resources/Alteration/CooperativeResource.ts (hunk 3 above)
Comment (lambert-alex) [reply src/resources/Alteration/CooperativeResource.ts]: I fixed all the ones I changed and made them correct sorry about that, that was a bone headed move.

File: src/interfaces/resource-interfaces/resource-interface.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (lambert-alex) [reply src/interfaces/resource-interfaces/resource-interface.ts]: I didn't add anything to the unit tests I didn't know how I could effectively check for a lack of staff payment option.

**Review by lewischenstudio — APPROVED** [review]

**Review by severinbeauvais — APPROVED** [review]
Looks good.

I just merged my PR with app version 3.1.12 so you should update to 13.

**Pull request merged**